src/commands/replay-sqs-dlq.js

```diff
@@ -1,42 +1,122 @@
+		} = flags;
 
 		global.region = region;
 		global.profile = profile;
 		global.keep = keep;
 
 		checkVersion();
 
-		this.log(`finding the queue [${dlqQueueName}] in [${region}]`);
+		this.log(`finding the SQS DLQ [${dlqQueueName}] in [${region}]`);
 		const dlqQueueUrl = await getQueueUrl(dlqQueueName);
 
-		this.log(`finding the queue [${queueName}] in [${region}]`);
-		const queueUrl = await getQueueUrl(queueName);
+		let sendToTarget;
+		switch (targetType) {
+			case "SNS":
+				this.log(`finding the SNS topic [${targetName}] in [${region}]`);
+				sendToTarget = this.sendToSNS(await getTopicArn(targetName));
+				break;
+			case "Kinesis":
+				sendToTarget = this.sendToKinesis(targetName);
+				break;
+			default:
+				this.log(`finding the SQS queue [${targetName}] in [${region}]`);
+				sendToTarget = this.sendToSQS(await getQueueUrl(targetName));
+				break;
+		}
 
 		this.log(
-			`replaying events from [${dlqQueueUrl}] to [${queueUrl}] with ${concurrency} concurrent pollers`
+			`replaying events from [${dlqQueueUrl}] to [${targetType}:${targetName}] with ${concurrency} concurrent pollers`
 		);
-		this.replay(dlqQueueUrl, queueUrl, concurrency);
+		await this.replay(dlqQueueUrl, concurrency, sendToTarget);
 
 		this.log("all done!");
 	}
 
-	async replay(dlqQueueUrl, queueUrl, concurrency) {
+	async replay(dlqQueueUrl, concurrency, sendToTarget) {
 		const promises = _.range(0, concurrency).map(() =>
-			this.runPoller(dlqQueueUrl, queueUrl)
+			this.runPoller(dlqQueueUrl, sendToTarget)
 		);
 		await Promise.all(promises);
 	}
 
-	async runPoller(dlqQueueUrl, queueUrl) {
+	sendToSQS(queueUrl) {
+		const AWS = getAWSSDK();
+		const SQS = new AWS.SQS();
+
+		return async messages => {
+			const sendEntries = messages.map(msg => ({
+				Id: msg.MessageId,
+				MessageBody: msg.Body,
+				MessageAttributes: msg.MessageAttributes
+			}));
+			await SQS.sendMessageBatch({
+				QueueUrl: queueUrl,
+				Entries: sendEntries
+			}).promise();
+		};
+	}
+
+	sendToSNS(topicArn) {
+		const AWS = getAWSSDK();
+		const SNS = new AWS.SNS();
+
+		return async messages => {
+			const promises = messages.map(async msg => {
+				const msgAttrPairs = _.toPairs(msg.MessageAttributes);
+				const supportedAttrs = msgAttrPairs
+					// eslint-disable-next-line no-unused-vars
+					.filter(([key, value]) => value.DataType === "String")
+					.map(([key, { DataType, StringValue }]) => [
+						key,
+						{ DataType, StringValue }
+					]);
+				const messageAttributes = _.fromPairs(supportedAttrs);
+				await SNS.publish({
+					Message: msg.Body,
+					MessageAttributes: messageAttributes,
+					TopicArn: topicArn
+				}).promise();
+			});
+
+			await Promise.all(promises);
+		};
+	}
+
+	sendToKinesis(streamName) {
+		const AWS = getAWSSDK();
+		const Kinesis = new AWS.Kinesis();
+
+		return async messages => {
+			const records = messages.map(msg => ({
+				Data: msg.Body,
+				PartitionKey: msg.MessageId
+			}));
+			await Kinesis.putRecords({
```

Comment:
Same as above. The limitation is 500. See https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html

Reply:
same as above